asdf/_tests/tags/core/tests/test_ndarray.py

```diff
@@ -378,17 +378,27 @@ def test_inline_bare():
         assert_array_equal(ff.tree["arr"], [[1, 2, 3, 4], [5, 6, 7, 8]])
 
 
-def test_mask_roundtrip(tmp_path):
-    x = np.arange(0, 10, dtype=float)
-    m = ma.array(x, mask=x > 5)
-    tree = {"masked_array": m, "unmasked_array": x}
+@pytest.mark.parametrize(
+    "mask",
+    [
+        [[False, False, True], [False, True, False], [False, False, False]],
+        True,
+        False,
+    ],
+)
+def test_mask_roundtrip(mask, tmp_path):
+    array = np.array([[1, 0, 0], [0, 1, 0], [0, 0, 0]])
+    tree = {
+        "unmasked": array,
+        "masked": np.ma.array(array, mask=mask),
+    }
 
     with roundtrip(tree) as af:
-        tree = af.tree
-
-        m = tree["masked_array"]
-
-        assert np.all(m.mask[6:])
+        # assert_array_equal ignores the mask, so use equality here
+        assert (tree["masked"].data == af["masked"].data).all()
+        assert (tree["masked"].mask == af["masked"].mask).all()
+        # ensure tree validity
+        assert (af["unmasked"] == af["masked"].data).all()
```

Comment:
I see now, my previous suggestion was wrong (it compared an array to itself). What's happening here is you're comparing the saved unmasked array to saved masked data to make sure they're equal (which they should be based on how the tree is set up). Makes sense now!

Reply:
yeah sorry! I was confused but initially thought you might just be doing some asdf wizardry that I didn't know about 😅